__tests__/taskRequests/taskRequest.test.js

```diff
@@ -0,0 +1,130 @@
+const puppeteer = require('puppeteer');
+
+const { fetchedTaskRequests } = require('../../mock-data/taskRequests');
+
+const SITE_URL = 'http://localhost:8000';
+// helper/loadEnv.js file causes API_BASE_URL to be stagin-api on local env url in taskRequest/index.html
+const API_BASE_URL = 'https://staging-api.realdevsquad.com';
+
+describe('Task Requests', () => {
+  let browser;
+  let page;
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch();
+    page = await browser.newPage();
+
+    await page.setRequestInterception(true);
+
+    page.on('request', (request) => {
+      if (request.url() === `${API_BASE_URL}/taskRequests`) {
+        request.respond({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify({ data: fetchedTaskRequests }),
+          headers: {
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
+            'Access-Control-Allow-Headers': 'Content-Type, Authorization',
+          },
+        });
+      } else {
+        request.continue();
+      }
+    });
+
+    await page.goto(`${SITE_URL}/taskRequests`);
+    await page.waitForNetworkIdle();
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
+
+  describe('When the user is super user', () => {
+    it('should display the task requests card', async () => {
+      const url = await page.evaluate(() => API_BASE_URL);
+      console.log(url);
+      const taskCards = await page.$$('.taskRequest__card');
+      const title = await taskCards[0].evaluate(
+        (el) => el.children[0].textContent,
+      );
+      const purpose = await taskCards[0].evaluate(
+        (el) => el.children[1].textContent,
+      );
```

Comment:
NIT: By accessing children in this way, later if the HTML structure changes then the test will fail

Reply:
can rewrite the code @prakashchoudhary07